### ISSUE TITLE
FTDI onboard does not have a unique serial number

### ISSUE THREAD

**Issue by lifeeth** [resin-io-playground/sd-mux#29]
AutoHAT v1 has FT245 - Chips sourced from the same batch produced right after each other do not have a unique serial number - we had to work around this by using the USB hubs serial number to identify a board. We reprogrammed the FT245's eeprom with the USB hubs unique serial number.

**Comment by lurch** [comment]
I guess this would be 'fixed' by the [FT245R](http://www.ftdichip.com/Products/ICs/FT245R.htm) ?

**Comment by lifeeth** [comment]
@lurch we do use a FT245R - which allows for us to reprogram the serial. We need to document this / have a process for this in the Readme

**Comment by lurch** [comment]
That product page says "A unique number (the FTDIChip-ID™) is burnt into the device during manufacture and is readable over USB" ?

**Comment by lifeeth** [comment]
@lurch unfortunately this isnt the Serial number used by the kernel though.

**Comment by lifeeth** [comment]
Even though http://www.ftdichip.com/Support/Documents/DataSheets/ICs/DS_FT245R.pdf says they have unique serial numbers - we had duplicates

**Comment by lurch** [comment]
http://www.ftdichip.com/Support/SoftwareExamples/FTDIChip-ID.htm says that it is accessible on Linux though, so maybe you'd need to temporarily unload the kernel driver to get the Chip-ID?

**Comment by lifeeth** [comment]
@lurch we rely on the serial number in https://github.com/resin-io/autohat/blob/master/libraries/RigControl.py - referencing by serial number makes it really easy and avoid a whole lot of software complexity - I am not sure what you are suggesting. We have a work around in place - this ticket is to see if there is a different approach to this - Maybe we will want to do away with the FTDI all together in the next AutoHAT version.

**Comment by lurch** [comment]
Fair enough. I guess programming the FT245 to also report the hub's serial number makes it easy to associate the relevant USB hub and FT245 devices together, in the event that you have more than one AutoHAT plugged into the same PC.

**Comment by lifeeth** [comment]
Exactly this is how we hit this issue - we have a rack in London with multiple AutoHATs connected to the same PC - we dont have any issues if have just one board connected to the PC
